src/nn/ops/pool2d.ts

```diff
@@ -16,31 +16,44 @@ export abstract class Pool extends SingleOutputOperation {
   protected dilations_?: [number, number];
   protected groups_?: number;
   protected layout_?: MLInputOperandLayout;
-  private autoPad_?: MLAutoPad;
   protected roundingType_?: MLRoundingType;
   protected outputSizes_?: [number, number];
   private needCheckOutputShape_ = true;
+  private outputShape_: [number, number, number, number];
 
   constructor(input: MLOperand, options: MLPooling2dOptions = {}) {
     super(input.builder);
     utils.validateOperand(input);
     this.input_ = input;
+
+    let windowDimensions: [number, number];
+    if (options.windowDimensions === undefined) {
+      if (options.layout === MLInputOperandLayout.nhwc) {
```

Comment:
May use switch case.

Reply:
Updated by optimizing code in new commit, thanks.